Assets/MRTK/Providers/Oculus/XRSDK/OculusXRSDKDeviceManager.cs

```diff
@@ -31,7 +35,14 @@ public class OculusXRSDKDeviceManager : XRSDKDeviceManager
             IMixedRealityInputSystem inputSystem,
             string name = null,
             uint priority = DefaultPriority,
-            BaseMixedRealityProfile profile = null) : base(inputSystem, name, priority, profile) { }
+            BaseMixedRealityProfile profile = null) : base(inputSystem, name, priority, profile)
+        {
+#if !OCULUSINTEGRATION_PRESENT && UNITY_EDITOR && UNITY_ANDROID
+            Debug.LogWarning(@"Be sure to download the Oculus Integration Package from the Unity Asset Store and run the Integration tool before deploying.
```

Comment:
Can this happen in an overridden `Initialize` method instead of in the constructor? It should still run in the editor.

Reply:
Does Initialize() run when the user tries to build the project? Something I noticed when trying to deploy/when helping alex deploy is that its easy for users to get to a step where they are trying to deploy to device and up until that point they experience no errors and end up confused at the vague deployment messages.

Reply:
Initialize() runs every time the MRTK object in the scene is reloaded, whether at edit-time or runtime.

Does this state (`!OCULUSINTEGRATION_PRESENT && UNITY_EDITOR && UNITY_ANDROID`) lead to deployment errors? It feels like this should fallback gracefully and still build for Android even without the integration (for example, if I'm building for ARCore).

Reply:
If we're trying to give messages at build time though, I don't think `InitializeOnLoad` is the way to go. That happens way more frequently, which is why we've been trying to reduce the amount of them.

Reply:
The state works as expected, I can try isolating controllers/hands a bit more, but the potential issue I highlighted regarding a reference to an unloaded Unity.XR.Oculus assembly still stands (I think it broke CI once). We can try again since that PR ended up fixing several issues if you think we can get away with it?

Reply:
Yeah, looks like we had the Unity.XR.Oculus reference checked-in for a bit and CI was fine with it.